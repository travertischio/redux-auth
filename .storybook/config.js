import React from 'react';
import PropTypes from 'prop-types';
import 'babel-polyfill';
import { fromJS } from 'immutable';
import {
  configure,
  addDecorator,
} from '@kadira/storybook';
import { IntlProvider } from 'react-intl';
import {
  createStore,
  applyMiddleware,
  compose,
} from 'redux';
import { combineReducers } from 'redux-immutable';
import createSagaMiddleware from 'redux-saga';
import {
  Provider,
  connect,
} from 'react-redux';
import { reducer as formReducer } from 'redux-form/immutable';
import { setBaseUrl } from 'api-client';
import { createStructuredSelector } from 'reselect';
import {
  AuthenticationProvider,
  reducer as authReducer,
  sagas as authSagas,
  selectIsAuthenticated,
} from '../src/index';
import signInPageReducer from '../src/containers/SignInPage/reducer';
import signInPageSagas from '../src/containers/SignInPage/sagas';
import signUpPageReducer from '../src/containers/SignUpPage/reducer';
import signUpPageSagas from '../src/containers/SignUpPage/sagas';
import resetPasswordPageReducer from '../src/containers/ResetPasswordPage/reducer';
import resetPasswordPageSagas from '../src/containers/ResetPasswordPage/sagas';
import requestPasswordResetPageReducer from '../src/containers/RequestPasswordResetPage/reducer';
import requestPasswordResetPageSagas from '../src/containers/RequestPasswordResetPage/sagas';
import signOutPageSagas from '../src/containers/SignOutPage/sagas';

setBaseUrl('https://demo-api-dev.arabel.la/api/');

const reducers = {
  form: formReducer,
  auth: authReducer,
  signInPage: signInPageReducer,
  signUpPage: signUpPageReducer,
  resetPasswordPage: resetPasswordPageReducer,
  requestPasswordResetPage: requestPasswordResetPageReducer,
};
const sagaMiddleware = createSagaMiddleware();
const reducer = combineReducers(reducers);
const store = createStore(
  reducer,
  fromJS({}),
  compose(
    applyMiddleware(sagaMiddleware),
    window.devToolsExtension ? window.devToolsExtension() : (f) => f
  )
);

authSagas.map(sagaMiddleware.run);
signInPageSagas.map(sagaMiddleware.run);
signUpPageSagas.map(sagaMiddleware.run);
signOutPageSagas.map(sagaMiddleware.run);
resetPasswordPageSagas.map(sagaMiddleware.run);
requestPasswordResetPageSagas.map(sagaMiddleware.run);

let IsAuthenticatedWrapper = (props) => {
  const {
    isAuthenticated,
    story,
  } = props;

  return (
    <main>
      {isAuthenticated && <h1>You are authenticated!</h1>}
      {!isAuthenticated && <h1>You are not authenticated!</h1>}

      <hr/>

      {story()}
    </main>
  );
};

IsAuthenticatedWrapper.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  story: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  isAuthenticated: selectIsAuthenticated,
});

IsAuthenticatedWrapper = connect(mapStateToProps)(IsAuthenticatedWrapper);

addDecorator((story) => (
  <IsAuthenticatedWrapper story={story} />
));

addDecorator((story) => (
  <AuthenticationProvider>
    {story()}
  </AuthenticationProvider>
));

addDecorator((story) => (
  <Provider store={store}>
    {story()}
  </Provider>
));

addDecorator((story) => (
  <IntlProvider locale="en">
    {story()}
  </IntlProvider>
));

const req = require.context('../src/', true, /\.stories\.js?$/);

function loadStories() {
  req.keys().forEach((filename) => req(filename));
}

configure(loadStories, module);
