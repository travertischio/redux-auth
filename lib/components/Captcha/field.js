'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createReduxFormField = require('react-form-fields/lib/createReduxFormField');

var _createReduxFormField2 = _interopRequireDefault(_createReduxFormField);

var _ = require('./');

var _2 = _interopRequireDefault(_);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _createReduxFormField2.default)(_2.default);