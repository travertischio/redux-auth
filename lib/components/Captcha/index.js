'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRecaptcha = require('react-recaptcha');

var _reactRecaptcha2 = _interopRequireDefault(_reactRecaptcha);

var _config = require('../../config');

var _config2 = _interopRequireDefault(_config);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                             *
                                                                                                                                                                                                                             * Captcha
                                                                                                                                                                                                                             *
                                                                                                                                                                                                                             */

var Captcha = function Captcha(_ref) {
  var onChange = _ref.onChange,
      rest = _objectWithoutProperties(_ref, ['onChange']);

  return _react2.default.createElement(_reactRecaptcha2.default, _extends({}, rest, {
    render: 'explicit',
    sitekey: _config2.default.recaptchaSiteKey,
    verifyCallback: onChange
  }));
};

Captcha.propTypes = {
  onChange: _propTypes2.default.func.isRequired
};

exports.default = Captcha;