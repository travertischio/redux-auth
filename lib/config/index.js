'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.setConfig = undefined;

var _creactUserIsNotAuthenticatedAuthWrapper = require('../hocs/AuthWrappers/creactUserIsNotAuthenticatedAuthWrapper');

var _creactUserIsNotAuthenticatedAuthWrapper2 = _interopRequireDefault(_creactUserIsNotAuthenticatedAuthWrapper);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var config = {
  userIsNotAuthenticatedRedirectPath: '/sign-in',
  userIsNotAdminRedirectPath: '/',
  userHasNoRoleRedirectPath: '/',
  userIsAuthenticatedRedirectPath: '/',
  redirectPathAfterSignIn: '/',
  redirectPathAfterSignUp: '/',
  redirectPathAfterSignOut: '/sign-in',
  adminRole: '20_example_admin',
  camelizeUserDataKeys: false
};

config.signInAuthWrapper = (0, _creactUserIsNotAuthenticatedAuthWrapper2.default)(config.redirectPathAfterSignIn);
config.signUpAuthWrapper = (0, _creactUserIsNotAuthenticatedAuthWrapper2.default)(config.redirectPathAfterSignUp);

var setConfig = exports.setConfig = function setConfig(newConfig) {
  Object.assign(config, newConfig);

  if (newConfig.redirectPathAfterSignIn) {
    updateSignInAuthWrapper();
  }

  if (newConfig.redirectPathAfterSignUp) {
    updateSignUpAuthWrapper();
  }
};

function updateSignInAuthWrapper() {
  config.signInAuthWrapper = (0, _creactUserIsNotAuthenticatedAuthWrapper2.default)(config.redirectPathAfterSignIn);
}

function updateSignUpAuthWrapper() {
  config.signUpAuthWrapper = (0, _creactUserIsNotAuthenticatedAuthWrapper2.default)(config.redirectPathAfterSignUp);
}

updateSignInAuthWrapper();
updateSignUpAuthWrapper();

exports.default = config;