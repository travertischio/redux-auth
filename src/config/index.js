const config = {
  userIsNotAuthenticatedRedirectPath: '/sign-in',
  userIsNotAdminRedirectPath: '/',
  userHasNoRoleRedirectPath: '/',
  userIsAuthenticatedRedirectPath: '/',
  redirectPathAfterSignOut: '/sign-in',
  signInConfirmCodePageUrl: '/sign-in-confirm-code',
  adminRole: '20_example_admin',
  camelizeUserDataKeys: true,
  reactRouterVersion: 3,
  autoSignOutWithin: false,
  signInAuthWrapper: undefined,
  signUpAuthWrapper: undefined,
  recaptchaSiteKey: undefined,
  successAuthenticationResponseSaga: undefined,
  failedAuthenticationResponseSaga: undefined,
};

export const setConfig = (newConfig) => {
  Object.assign(config, newConfig);
};

export default config;
