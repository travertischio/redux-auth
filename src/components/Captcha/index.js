/**
*
* Captcha
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import Recaptcha from 'react-recaptcha';
import config from '~/config';

const Captcha = ({ onChange, ...rest }) => (
  <Recaptcha
    {...rest}
    render="explicit"
    sitekey={config.recaptchaSiteKey}
    verifyCallback={onChange}
  />
);

Captcha.propTypes = {
  onChange: PropTypes.func.isRequired,
};

export default Captcha;
