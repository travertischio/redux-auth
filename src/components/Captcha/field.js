import createReduxFormField from 'react-form-fields/lib/createReduxFormField';
import Captcha from './';

export default createReduxFormField(Captcha);
