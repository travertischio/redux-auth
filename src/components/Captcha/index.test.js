import React from 'react';
import { createComponentWithIntl } from 'react-unit-testing-utils';
import { setConfig } from '~/config';
import Captcha from './';
import CaptchaField from './field';

describe('<Captcha />', () => {
  it('should render Captcha', () => {
    const initialState = {};
    const { component } = createComponentWithIntl(<Captcha onChange={() => true} />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render Captcha with siteKey set by config', () => {
    setConfig({ recaptchaSiteKey: 'ABC1234XYZ' });
    const initialState = {};
    const { component } = createComponentWithIntl(<Captcha onChange={() => true} />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render CaptchaField', () => {
    const initialState = {};
    const { component } = createComponentWithIntl(<CaptchaField input={{ onChange: () => true }} meta={{}} />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
