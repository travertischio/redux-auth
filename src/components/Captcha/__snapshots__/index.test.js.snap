// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Captcha /> should render Captcha 1`] = `
<div
  className="g-recaptcha"
  data-badge="bottomright"
  data-sitekey={undefined}
  data-size="normal"
  data-tabindex="0"
  data-theme="light"
  data-type="image"
  id="g-recaptcha"
/>
`;

exports[`<Captcha /> should render Captcha with siteKey set by config 1`] = `
<div
  className="g-recaptcha"
  data-badge="bottomright"
  data-sitekey="ABC1234XYZ"
  data-size="normal"
  data-tabindex="0"
  data-theme="light"
  data-type="image"
  id="g-recaptcha"
/>
`;

exports[`<Captcha /> should render CaptchaField 1`] = `
<div
  className=""
>
  <div
    className="g-recaptcha"
    data-badge="bottomright"
    data-sitekey="ABC1234XYZ"
    data-size="normal"
    data-tabindex="0"
    data-theme="light"
    data-type="image"
    id="g-recaptcha"
  />
</div>
`;
